models/record.go

```diff
@@ -2,35 +2,61 @@ package models
 
 import (
 	"regexp"
-
-	"github.com/pborman/uuid"
 )
 
 type Record struct {
 	Person
+	RecordGroup
 	MedicalRecordNumber   string                `json:"medical_record_number,omitempty"`
 	MedicalRecordAssigner string                `json:"medical_record_assigner,omitempty"`
 	BirthDate             int64                 `json:"birthdate,omitempty"`
 	DeathDate             int64                 `json:"deathdate,omitempty"`
 	Expired               bool                  `json:"expired,omitempty"`
-	Encounters            []Encounter           `json:"encounters,omitempty"`
-	LabResults            []LabResult           `json:"results,omitempty"`
-	Languages             []Language            `json:"languages,omitempty"`
-	ProviderPerformances  []ProviderPerformance `json:"provider_performances,omitempty"`
-	InsuranceProviders    []InsuranceProvider   `json:"insurance_providers,omitempty"`
-	Procedures            []Procedure           `json:"procedures,omitempty"`
-	Medications           []Medication          `json:"medications, omitempty"`
-	Allergies             []Allergy             `json:"allergies,omitempty"`
-	Conditions            []Condition           `json:"conditions,omitempty"`
-	Communications        []Communication       `json:"communications,omitempty"`
-	MedicalEquipment      []MedicalEquipment    `json:"medical_equipment,omitempty"`
-	CareGoals             []Entry               `json:"care_goals,omitempty"`
+	// Private values to handle querying of Entries
+	
+}
+
+
+type RecordGroup struct {
+	Encounters            EncounterGroup           `json:"encounters,omitempty"`
+	LabResults            LabResultsGroup           `json:"results,omitempty"`
+	ProviderPerformances  ProviderPerformanceGroup `json:"provider_performances,omitempty"`
+	InsuranceProviders    InsuranceProviderGroup   `json:"insurance_providers,omitempty"`
+	Procedures            ProcedureGroup           `json:"procedures,omitempty"`
+	Medications           MedicationGroup          `json:"medications, omitempty"`
+	Allergies             AllergyGroup             `json:"allergies,omitempty"`
+	Conditions            ConditionGroup           `json:"conditions,omitempty"`
+	
+	// These weren't in the Entries() method.
+	Languages             LanguagesGroup            `json:"languages,omitempty"`
+	Communications        CommunicationGroup       `json:"communications,omitempty"`
+	MedicalEquipment      MedicalEquipmentGroup    `json:"medical_equipment,omitempty"`
+	CareGoals             EntryGroup               `json:"care_goals,omitempty"`
 }
 
 type Language struct {
 	Coded
 }
 
+type EntryGroup []Entry
```

Comment:
I'm not taking advantage of the types yet, but they are there because they will eventually implement the interface described below them (also not being used yet). This way the particular logic in the nest of if statements in EntriesForOids will get distributed into the correct function on that type.